include/RAJA/index/RangeSegment.hpp

```diff
@@ -87,8 +97,9 @@ struct TypedRangeSegment {
    * \param[in] begin the starting value (inclusive) for the range
    * \param[in] end the ending value (exclusive) for the range
    */
-  RAJA_HOST_DEVICE constexpr TypedRangeSegment(DiffT begin, DiffT end)
-      : m_begin(iterator{begin}), m_end(iterator{end})
+  using StripStorageT = strip_index_type_t<StorageT>;
+  RAJA_HOST_DEVICE constexpr TypedRangeSegment(StripStorageT begin, StripStorageT end)
```

Comment:
Should these arguments be the stripped type, or the actual storage type? @ajkunen

Reply:
the iterator expects to receive a stripped type (it was previously receiving a DiffT value), therefore using RAJA generated types such as those created by RAJA_INDEX_VALUE etc. causes errors

Reply:
That makes sense to me. I wonder if we should strip them internally though, rather than accepting the stripped type. I'd be interested in what Adam thinks.